### ISSUE TITLE
guild publish -E returns FileNotFoundError

### ISSUE THREAD

**Issue by gryBox** [guildai/guildai#22]
Hi @gar1t  - I am using the `pre` build of `guild` on a aws linux os in a python 3.6 conda virtual environment.  When I run `guild publish -E` I get the following error:
`FileNotFoundError: [Errno 2] No such file or directory: '/home/ec2-user/anaconda3/envs/py36-ml/.guild/runs/43b3b7e87e9f11e9b5500a1e73c110c6/441f56927e9f11e996230a1e73c110c6'`

**Comment by gar1t** [comment]
Thanks for the report!

Could you verify that you're seeing this traceback?

P.S. If you get a traceback for an error, it's extremely helpful in debugging an issue - just paste it into the issue if it's available.

```
Publishing [146fb9f4] test+ 2019-05-28 06:09:36 error 
Traceback (most recent call last):
  File "/home/garrett/SCM/guild/guild/scripts/guild", line 30, in <module>
    main()
  File "/home/garrett/SCM/guild/guild/scripts/guild", line 23, in main
    guild.main_bootstrap.main()
  File "/home/garrett/SCM/guild/guild/main_bootstrap.py", line 48, in main
    guild.main.main()
  File "/home/garrett/SCM/guild/guild/main.py", line 32, in main
    _main()
  File "/home/garrett/SCM/guild/guild/main.py", line 38, in _main
    main_cmd.main(standalone_mode=False)
  File "/home/garrett/SCM/guild/guild/external/click/core.py", line 759, in __call__
    return self.main(*args, **kwargs)
  File "/home/garrett/SCM/guild/guild/external/click/core.py", line 714, in main
    rv = self.invoke(ctx)
  File "/home/garrett/SCM/guild/guild/external/click/core.py", line 1132, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/garrett/SCM/guild/guild/external/click/core.py", line 951, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/garrett/SCM/guild/guild/external/click/core.py", line 552, in invoke
    return callback(*args, **kwargs)
  File "/home/garrett/SCM/guild/guild/external/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/garrett/SCM/guild/guild/click_util.py", line 146, in fn
    return fn0(*(args + (Args(**kw),)))
  File "/home/garrett/SCM/guild/guild/commands/publish.py", line 34, in publish
    runs_impl.publish(args, ctx)
  File "/home/garrett/SCM/guild/guild/commands/runs_impl.py", line 756, in publish
    _publish(args, ctx)
  File "/home/garrett/SCM/guild/guild/commands/runs_impl.py", line 769, in _publish
    publish_f, ALL_RUNS_ARG, True)
  File "/home/garrett/SCM/guild/guild/commands/runs_impl.py", line 325, in _runs_op
    op_callback(selected, formatted)
  File "/home/garrett/SCM/guild/guild/commands/runs_impl.py", line 765, in publish_f
    _publish_runs(runs, formatted, args)
  File "/home/garrett/SCM/guild/guild/commands/runs_impl.py", line 785, in _publish_runs
    formatted_run=fmt)
  File "/home/garrett/SCM/guild/guild/publish.py", line 252, in publish_run
    template.generate(run_dest, render_vars)
  File "/home/garrett/SCM/guild/guild/publish.py", line 199, in generate
    _render_template(template, vars, file_dest)
  File "/home/garrett/SCM/guild/guild/publish.py", line 229, in _render_template
    for part in template.generate(vars):
  File "/home/garrett/.local/lib/python2.7/site-packages/jinja2/environment.py", line 1045, in generate
    yield self.environment.handle_exception(exc_info, True)
  File "/home/garrett/.local/lib/python2.7/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/garrett/SCM/guild/guild/templates/publish-default/README.md", line 5, in top-level template code
    {% block body %}
  File "/home/garrett/SCM/guild/guild/templates/publish-default/README.md", line 28, in block "body"
    {% block files %}
  File "/home/garrett/SCM/guild/guild/templates/publish-default/README.md", line 31, in block "files"
    {% include ["_files.md", "publish-default/_files.md"] %}
  File "/home/garrett/SCM/guild/guild/templates/publish-default/_files.md", line 1, in top-level template code
    {% set run_files = run|files -%}
  File "/home/garrett/SCM/guild/guild/publish.py", line 100, in _files
    return _format_run_files(run["_run"])
  File "/home/garrett/SCM/guild/guild/publish.py", line 163, in _format_run_files
    mtime = os.path.getmtime(abspath)
  File "/usr/lib/python2.7/genericpath.py", line 62, in getmtime
    return os.stat(filename).st_mtime
OSError: [Errno 2] No such file or directory: '/home/garrett/.guild/runs/146fb9f4813911e9b16de4a471939b0d/14856fba813911e9b16de4a471939b0d'
```

**Comment by gryBox** [comment]
Traceback:
```
Traceback (most recent call last):
  File "/home/ec2-user/anaconda3/envs/py36-ml/bin/guild", line 10, in <module>
    sys.exit(main())
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/main_bootstrap.py", line 48, in main
    guild.main.main()
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/main.py", line 32, in main
    _main()
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/main.py", line 38, in _main
    main_cmd.main(standalone_mode=False)
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/external/click/core.py", line 759, in __call__
    return self.main(*args, **kwargs)
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/external/click/core.py", line 714, in main
    rv = self.invoke(ctx)
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/external/click/core.py", line 1132, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/external/click/core.py", line 951, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/external/click/core.py", line 552, in invoke
    return callback(*args, **kwargs)
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/external/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/click_util.py", line 146, in fn
    return fn0(*(args + (Args(**kw),)))
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/commands/publish.py", line 34, in publish
    runs_impl.publish(args, ctx)
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/commands/runs_impl.py", line 756, in publish
    _publish(args, ctx)
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/commands/runs_impl.py", line 769, in _publish
    publish_f, ALL_RUNS_ARG, True)
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/commands/runs_impl.py", line 325, in _runs_op
    op_callback(selected, formatted)
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/commands/runs_impl.py", line 765, in publish_f
    _publish_runs(runs, formatted, args)
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/commands/runs_impl.py", line 785, in _publish_runs
    formatted_run=fmt)
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/publish.py", line 252, in publis _run
    template.generate(run_dest, render_vars)
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/publish.py", line 199, in generate
    _render_template(template, vars, file_dest)
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/publish.py", line 229, in _render_template
    for part in template.generate(vars):
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/jinja2/environment.py", line 1045, in generate
    yield self.environment.handle_exception(exc_info, True)
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/templates/publish-default/README.md", line 5, in top-level template code
    {% block body %}
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/templates/publish-default/README.md", line 28, in block "body"
    {% block files %}
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/templates/publish-default/README.md", line 31, in block "files"
    {% include ["_files.md", "publish-default/_files.md"] %}
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/templates/publish-default/_files.md", line 1, in top-level template code
    {% set run_files = run|files -%}
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/publish.py", line 100, in _files
    return _format_run_files(run["_run"])
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/publish.py", line 163, in _format_run_files
    mtime = os.path.getmtime(abspath)
  File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/genericpath.py", line 55, in getmtime
    return os.stat(filename).st_mtime
FileNotFoundError: [Errno 2] No such file or directory: '/home/ec2-user/anaconda3/envs/py36-ml/.guild/runs/43b3b7e87e9f11e9b5500a1e73c110c6/441f56927e9f11e996230a1e73c110c6'
```

**Comment by gar1t** [comment]
Hah,a yep that's the one :) Thanks! Fix in 9d905ae

**Comment by gar1t** [comment]
Closing - thanks again for the report!
